Source/ORTS.Common/VirtualFileSystem.cs

```diff
@@ -0,0 +1,717 @@
+                            MountArchive(archivePath, subPath, mountpoint);
+                            return true;
+                        }
+                        else
+                        {
+                            message = $"VFS mount: Source archive format is not supported, skipping: {source} => {mountpoint}";
+                            Trace.TraceWarning(message);
+                            InitLog.Enqueue(message);
+                            return false;
+                        }
+                    }
+                    if (!Directory.Exists(archivePath) || i == sourcePieces.Length - 2)
+                    {
+                        // During the search we diverged already to a non-existent directory, we can safely abort.
+                        // Or we reached the last searchable part, and still not found the base archive.
+                        message = $"VFS mount: Source file not found, skipping: {source} => {mountpoint}";
+                        Trace.TraceWarning(message);
+                        InitLog.Enqueue(message);
+                        return false;
+                    }
+                }
+            }
+            message = $"VFS mount: Cannot parse configuration line, skipping: {source} => {mountpoint}";
+            Trace.TraceWarning(message);
+            InitLog.Enqueue(message);
+            return false;
+        }
+
+        static void MountDirectory(string directory, string mountpoint)
+        {
+            Debug.Assert(VfsRoot != null, "VFS is uninitialized");
+
+            var message = $"VFS mounting directory: {directory} => {mountpoint}";
+            Trace.TraceInformation(message);
+            InitLog.Enqueue(message);
+
+            var mountNode = VfsRoot.ChangeDirectory(mountpoint, true);
+            mountNode.SetDirectoryWritable(directory);
+
+            Stack.Clear();
+            Stack.Push((directory, mountNode));
+            while (Stack.Count > 0)
+            {
+                var (dirpath, vfsNode) = Stack.Pop();
+                foreach (var subdir in Directory.GetDirectories(dirpath))
+                {
+                    var name = NormalizeVirtualPath(Path.GetFileName(subdir.TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar)));
+                    Stack.Push((subdir, vfsNode.CreateDirectory(name, subdir)));
+                }
+                foreach (var file in Directory.GetFiles(dirpath))
+                {
+                    if (IsArchiveSupported(file))
+                    {
+                        if (NoAutoMount)
+                        {
+                            message = $"VFS skipped auto-mounting archive by user settings: {file}";
+                            Trace.TraceInformation(message);
+                            InitLog.Enqueue(message);
+                        }
+                        else
+                            MountArchive(file, null, vfsNode.GetVfsPath());
```

Comment:
Hang on, is this going to mount every zip file in a content folder **recursively**? That's not a thing we want to happen by default, and possibly at all

Reply:
My idea was not only to support future well-defined and well-built zip packages, but also the 20 years log inheritance we have. To me it was convenient to just drom a downloaded zip into subdirectory, and get that auto-mounted. The only case when this cannot be done, is when the zip has some inside fantasy-named subfolder that I didn't want to use. In that case registering that into the vfs-config file is the only choice. In short, the purpose of doing it recursively was to be able to auto-mount a zip that is "too short" in paths inside.

Reply:
Okay, so I'm not against recursively looking for ZIPs to mount, but it must be opt-in IMHO